app/helpers/job_invocation_output_helper.rb

```diff
@@ -0,0 +1,34 @@
+module JobInvocationOutputHelper
+  def colorize_line(line)
+    line = line.gsub(/\e\[.*?m/) { |seq| color = seq[/(\d+)m/,1]; "{{{format color:#{color}}}}" }
+
+    @current_color ||= 'default'
+    out = %{<span style="color: #{@current_color}">}
+    parts = line.split(/({{{format.*?}}})/)
+    parts.each do |line|
+      if line.include?('{{{format')
+        if color_index = line[/color:(\d+)/, 1]
+          @current_color = case color_index
```

Comment:
How about putting the mapping into a deeply frozen hash constant and just accessing it here to split it up a bit?